### PR TITLE
Make sure the control plane can be inspected by 'istioctl verify-install'

### PR DESCRIPTION
**Pull request by esnible**
Resolves https://github.com/istio/istio/issues/26265 "Test to make sure 'verify-install' works"
Resolves https://github.com/istio/istio/issues/26026 "istioctl verify-install fails on K8s 1.18 due to managedTime field

Also fixes `--manifests` not working in conjunction with `-f`.
Fixes deployment count by including "istiod" and "istiod-*"

### CODE REVIEW

**Comment by howardjohn** [comment]
You linked 7 open bugs, is this expected to fail?

**Comment by esnible** [comment]
> You linked 7 open bugs, is this expected to fail?

@howardjohn Earlier I forgot to tag this as `do-not-merge/work-in-progress`.

This PR does two things.  First, it fixes two problems with `istioctl verify-install`.  Second it adds `verify-install` to every Integration Istio deployment operation.

Let me know if you would prefer to see the fixes and additional testing put in the same or different PRs.
There is no way to flag adding verify-install to be skipped.  It runs on every install.  Let me know if you would prefer a `verify-install` specific test instead.

**Comment by esnible** [comment]
@ostromart I am willing to split this PR.  It does four things that I discovered together trying to make sense of all of the reports of problems with `verify-install`.

The most important is that it fixes problems with `managedFields` under Kubernetes 1.18.  The problem again is gogo library and "time".  What I did for that was to refactor the pre-gogo cleanup into a method `cleanupForGogo` and clear that field as Istio doesn't need it.

The next thing it does is allows `--manifests` to work with `-f`.  It wasn't working before.  This is a problem because I was getting IOPs with `installPackagePath: /home/prow/go/src/istio.io/istio/manifests` and I was running `istioctl` on a machine that doesn't have a `prow` user.  This fix may not be enough; should we go further and ignore `installPackagePath` when doing `verify-install`, which may be on a different machine?

The next thing it does is correct the count of Istio deployments.  I believe some Istio deployments are now named `istiod-*`, and it expecting `istio-*`.

In addition to these fixes I modified the test library so that tries 'verify-install' after installing.  This will catch problems and keep us from getting into this situation again.

File: istioctl/pkg/install/verify.go [hunk 1]
```diff
@@ -182,13 +184,17 @@ func verifyPostInstall(enableVerbose bool, istioNamespaceFlag string,
 			// usual conversion not available.  Convert unstructured to string
 			// and ask operator code to unmarshal.
 
-			un.SetCreationTimestamp(meta_v1.Time{}) // UnmarshalIstioOperator chokes on these
+			cleanupForGogo(un)
```
Comment (ostromart) [inline istioctl/pkg/install/verify.go:187]: is this gogo specific? will it work if gogo is replaced? if yes, i suggest removing gogo from the name and just do whatever is needed.

File: istioctl/pkg/install/verify.go [hunk 2]
```diff
@@ -453,3 +461,16 @@ func allOperatorsInCluster(client dynamic.Interface) ([]*v1alpha1.IstioOperator,
...
+// cleanupForGogo prepares an unstructed for github.com/gogo/protobuf's Unmarshal()
+func cleanupForGogo(un *unstructured.Unstructured) {
+	un.SetCreationTimestamp(meta_v1.Time{})             // operator_istio.UnmarshalIstioOperator chokes on these
+	un.SetManagedFields([]meta_v1.ManagedFieldsEntry{}) // operator_istio.UnmarshalIstioOperator chocks on the time in these
+}
+
+// isDeploymentName tells if this is a Deployment name created by "istioctl install" for the control plane
+func isDeploymentName(name string) bool {
+	return strings.HasPrefix(name, "istio-") ||
+		name == "istiod" || // 1.7.0 beta 1 uses this for master Istiod
```
Comment (ostromart) [inline istioctl/pkg/install/verify.go:474]: i feel uneasy about these sorts of tests being buried deep in the code. is it possible to make this more visible by declaring a const somewhere prominent? i also wonder if it's feasible to write a test to catch if something in the naming changes.

File: istioctl/pkg/install/verify.go (hunk 1 above)
Comment (esnible) [reply istioctl/pkg/install/verify.go:187]: The function is a work around for what is either a bug in Gogo or in Kubernetes.  This prevents `Error: could not load IstioOperator from cluster: could not unmarshal: json: cannot unmarshal string into Go value of type map[string]json.RawMessage`.  I suspect it is related to https://github.com/istio/istio/issues/26010 but I haven't proved that yet.

**Comment by esnible** [comment]
I created https://github.com/istio/istio/pull/26308 which holds only the fixes, not the new test.  We can merge that one separately, as I want that one in beta 2.  After that one merges I will revise this one.  For now this one is on `hold`.

**Pull request closed without merging**